modules/@apostrophecms/schema/ui/apos/mixins/AposInputChoicesMixin.js

```diff
@@ -29,5 +29,24 @@ export default {
     } else {
       this.choices = this.field.choices;
     }
+
+    if (this.field.type === 'select') {
+      this.prependEmptyChoice();
+    }
```

Comment:
moved here to prepend the empty choice after async results have returned, in case of dynamic choices